src/Illuminate/Filesystem/FilesystemAdapter.php

```diff
@@ -328,13 +328,25 @@ public function url($path)
      *
      * @param  \League\Flysystem\AwsS3v3\AwsS3Adapter  $adapter
      * @param  string  $path
+     * @param  string  $expires
```

Comment:
Can also be an `int` or `DateTime`, you could also convert carbon instances into one of these and pass it through :)

Source: https://github.com/aws/aws-sdk-php/blob/33e8f7083fcb54fd926a8569cb81a0795e8c2633/src/S3/S3ClientInterface.php#L17